File: Program31.c
#include<stdio.h>

void DisplayFactors(int iNo)
{
    int iCnt = 0 ;

    printf("Factors of %d are :\n",iNo);

    for(iCnt = 1 ; iCnt <= iNo ; iCnt++ ) //Time Complexity O(N):order of N
    {
        if((iNo % iCnt) == 0)
        {
            printf("%d\n",iCnt);
        }       
    }
}

int main()
{
    int iValue = 0 ;

    printf("Enter number : \n");
    scanf("%d",&iValue);
    
    DisplayFactors(iValue);

    return 0 ;
}

File: Program31.py
# Accept N Numbers from user and a Number , and check whether that Number is present or not 

def checkOccurrence(brr , iValue):
    bFlag = False 

    for i in brr :  
        if ( i == iValue):
            bFlag = True
            break

    return bFlag 

def main():
    iLength = 0 
    iNo = 0 

    iLength = int(input("Enter the size of Array :  "))
    iNo = int(input("Enter a Number : "))  

    Arr = list() 

    print("Enter the Elements of Array : ") 

    for i in range(iLength):
        Arr.append(int(input())) 

    bRet = checkOccurrence(Arr,iNo) 

    if(bRet == True ):
        print("Present...") 
    else :
        print("Absent")

if __name__=="__main__":
    main()
 